resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -418,17 +417,34 @@ private[spark] class MesosClusterScheduler(
     envBuilder.build()
   }
 
+  private def isContainerLocalAppJar(desc: MesosDriverDescription): Boolean = {
+    val isLocalJar = desc.jarUrl.startsWith("local://")
+    val isContainerLocal = desc.conf.getOption("spark.mesos.appJar.local.resolution.mode").exists {
```

Comment:
instead of `.exists` shouldn't this be `.flatmap`?

also does `case other` handles the case when the option is unset?

Reply:
Not really. Option.exists() returns false when there the option has vale None.
Other describes an arbitrary string here. Target here is to return a Boolean value and exists does just that under certain conditions.

Reply:
interesting!